app/src/main/java/unipd/se18/ocrcamera/TextExtractor.java

```diff
@@ -0,0 +1,240 @@
+        return firebaseVisionTextTask.getResult();
+    }
+
+    /**
+     * Produces a String from a FirebaseVisionText
+     * More details at: {@link FirebaseVisionText.TextBlock#getText()}.
+     * @param firebaseVisionTextExtracted The result of the FirebaseVisionText extraction
+     * @return String extracted by the FirebaseVisionText result
+     * @author Pietro Prandini (g2)
+     */
+    private String extractString(FirebaseVisionText firebaseVisionTextExtracted) {
+        // Orders the blocks (requested by the issue #18 but not so useful at the moment)
+        ArrayList<FirebaseVisionText.TextBlock> textBlocks =
+                sortBlocks(firebaseVisionTextExtracted);
+
+        // Prepares the String with the text of the blocks identified by the Firebase process
+        StringBuilder text = new StringBuilder();
+        for(FirebaseVisionText.TextBlock block: textBlocks) {
+            text.append(block.getText()).append("\n");
+        }
+        String extractedText = text.toString();
+
+        // Returns the String
+        return extractedText;
+    }
+
+    /*
+    Sorting methods are not so useful at the moment.
+    There is an issue (#18) that request this method for a possible future use.
+     */
+
+    /**
+     * Sorts the blocks recognized
+     * More details at: {@link FirebaseVisionText.TextBlock#getTextBlocks()}.
+     * @param OCRResult FirebaseVisionText object produced by an OCR recognition
+     * @return An ArrayList of FirebaseVisionText sorted
+     * @author Pietro Prandini (g2)
+     */
+    private ArrayList<FirebaseVisionText.TextBlock> sortBlocks(FirebaseVisionText OCRResult) {
+        ArrayList<FirebaseVisionText.TextBlock> OCRBlocks =
+                new ArrayList<>(OCRResult.getTextBlocks());
+        OCRBlocks = sortBlocksY(OCRBlocks);
+        return OCRBlocks;
+    }
+
+    /**
+     * Sorts the blocks recognized from top to bottom
+     * More details at: {@link FirebaseVisionText.TextBlock#getCornerPoints()}, {@link Comparator},
+     * {@link Collections#sort(List, Comparator)}.
+     * @param OCRBlocks ArrayList of FirebaseVisionText.TextBlock recognized by the OCR processing
+     * @return An ArrayList of FirebaseVisionText sorted from top to bottom
+     * @author Pietro Prandini (g2)
+     */
+    private ArrayList<FirebaseVisionText.TextBlock>
+    sortBlocksY(ArrayList<FirebaseVisionText.TextBlock> OCRBlocks) {
+        // Comparator for ordering the blocks by the y axis
+        Comparator<FirebaseVisionText.TextBlock> mYComparator =
+                new Comparator<FirebaseVisionText.TextBlock>() {
+            @Override
+            public int compare(FirebaseVisionText.TextBlock o1, FirebaseVisionText.TextBlock o2) {
+                int o1TopLeftY = Objects.requireNonNull(o1.getCornerPoints())[TOP_LEFT].y;
+                int o2TopLeftY = Objects.requireNonNull(o2.getCornerPoints())[TOP_LEFT].y;
+                return Integer.compare(o1TopLeftY,o2TopLeftY);
+            }
```

Comment:
I do not understand why you compare only the blocks on the top left. Would not it be better to compare all blocks?